### PR TITLE
add llm evaluate for language modeling

### PR DESCRIPTION
**Pull request by xin3he**
## Type of Change

example

## Description

migrate examples of LLMs in language modeling from ITREX to INC

## Expected Behavior & Potential Risk

extention test pass

### CODE REVIEW

File: examples/pytorch/nlp/huggingface_models/language-modeling/quantization/ptq_static/ipex/README.md [hunk 1]
```diff
@@ -0,0 +1,202 @@
...
+The script `run_clm_no_trainer.py` supports `GPTJ`, `OPT`, `LLaMA`, `BLOOM`, `MPT` and `Falcon` quantization and validates last word prediction accuracy with [lm_eval](https://github.com/EleutherAI/lm-evaluation-harness.git) now, and we are adding more models.
+
+# Prerequisite
+## 1. Create Environment
+```
+# Installation
+git clone https://github.com/intel/intel-extension-for-transformers.git itrex
+cd itrex
+pip install -r requirements.txt
+SKIP_RUNTIME=True pip install -e .
```
Comment (hshen14) [inline examples/pytorch/nlp/huggingface_models/language-modeling/quantization/ptq_static/ipex/README.md:14]: please install itrex binary directly. SKIP_RUNTIME is for dev mode

File: examples/pytorch/nlp/huggingface_models/language-modeling/quantization/ptq_static/ipex/run_quant.sh [hunk 2]
```diff
@@ -0,0 +1,84 @@
...
+        model_name_or_path="facebook/opt-125m"
+        extra_cmd=$extra_cmd" --ipex --int8_bf16_mixed"
+    elif [ "${topology}" = "opt_125m_ipex_sq" ]; then
+        model_name_or_path="facebook/opt-125m"
+        extra_cmd=$extra_cmd" --ipex --int8_bf16_mixed --sq --alpha 0.5"
+    elif [ "${topology}" = "llama_7b_ipex_sq" ]; then
+        model_name_or_path="decapoda-research/llama-7b-hf"
+        extra_cmd=$extra_cmd" --ipex --sq --alpha auto"
+    elif [ "${topology}" = "gpt_j_ipex_sq" ]; then
+        model_name_or_path="/tf_dataset2/models/pytorch/gpt-j-6B"
```
Comment (chensuyue) [inline examples/pytorch/nlp/huggingface_models/language-modeling/quantization/ptq_static/ipex/run_quant.sh:67]: model_name_or_path="${input_model}"

File: examples/pytorch/nlp/huggingface_models/language-modeling/quantization/ptq_static/ipex/requirements.txt [hunk 3]
```diff
@@ -0,0 +1,13 @@
...
+sentencepiece != 0.1.92
+datasets >= 1.1.3
+torch >= 1.10
+transformers
+pytest
+wandb
+einops
+neural-compressor
+intel-extension-for-transformers
+git+https://github.com/EleutherAI/lm-evaluation-harness.git@83dbfbf6070324f3e5872f63e49d49ff7ef4c9b3
```
Comment (wenhuach21) [inline examples/pytorch/nlp/huggingface_models/language-modeling/quantization/ptq_static/ipex/requirements.txt:12]: this version is outdated, we have observed significant changes in some results with the latest version.

**Review by wenhuach21 — APPROVED** [review]

**Comment by changwangss** [comment]
1. please remove smoothquant, the model generated by this scripte couldn't get the well performance due to past_key_values missing.
2. please also change the folder "ipex"

File: examples/pytorch/nlp/huggingface_models/language-modeling/quantization/ptq_static/ipex/run_benchmark.sh [hunk 4]
```diff
@@ -0,0 +1,111 @@
...
+        approach="weight_only"
+        extra_cmd=$extra_cmd" --woq_algo TEQ"
+    elif [ "${topology}" = "opt_125m_ipex" ]; then
+        model_name_or_path="facebook/opt-125m"
+        extra_cmd=$extra_cmd" --ipex --int8_bf16_mixed"
+    elif [ "${topology}" = "opt_125m_ipex_sq" ]; then
+        model_name_or_path="facebook/opt-125m"
+        extra_cmd=$extra_cmd" --ipex --int8_bf16_mixed --sq --alpha 0.5"
+    elif [ "${topology}" = "llama_7b_ipex_sq" ]; then
+        model_name_or_path="decapoda-research/llama-7b-hf"
```
[the remark this answers is not in the document]
Comment (wenhuach21) [reply examples/pytorch/nlp/huggingface_models/language-modeling/quantization/ptq_static/ipex/run_benchmark.sh:93]: https://huggingface.co/huggyllama/llama-7b,  we are using this one from our side. Xin, you may have a try.

**Comment by YIYANGCAI** [comment]
I have merged the general GPTQ's example in run_clm_no_trainer.py, and mlperf related files are also ported into this folder.

**Review by chensuyue — APPROVED** [review]
Extension test pass.

**Pull request merged**